main_routers/system_router.py

```
@@ -0,0 +1,483 @@
+                
+                if not workshop_related_dir:
+                    content_match = re.search(r'(.*?steamapps[/\\]workshop[/\\]content)', decoded_path, re.IGNORECASE)
+                    if content_match:
+                        workshop_related_dir = content_match.group(1)
+                
+                if workshop_related_dir and os.path.exists(workshop_related_dir):
+                    real_workshop_dir = os.path.realpath(workshop_related_dir)
+                    if real_workshop_dir not in allowed_dirs:
+                        allowed_dirs.append(real_workshop_dir)
+                        logger.info(f"动态添加允许的创意工坊相关目录: {real_workshop_dir}")
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)